tests/unit/s2n_sike_p503_kat_test.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * Modified from PQCgenKAT_kem.c
+ * Created by Bassham, Lawrence E (Fed) on 8/29/17.
+ * Copyright © 2017 Bassham, Lawrence E (Fed). All rights reserved.
+ */
+
+#include "s2n_test.h"
+#include "s2n_nist_kats.h"
+#include "pq-crypto/sike/sike_p503_kem.h"
+#include "pq-crypto/pq-random.h"
+
+#define RSP_FILE_NAME "kats/sike_p503.kat"
```

Comment:
NP: did you think about what a generic KAT test runner would look like? I was thinking if it would be cool to make a generic test that takes a file name, the 3 function pointers, and the sizes.

Reply:
Yes, I already moved the utility functions to an external common header. I left the main flow as is (duplicated) because of two reasons:
  1) We only have two KAT tests (SIKE p503/BIKE1_L1)
  2) Passing function pointers, in some cases might complicate things and will need a special care.
In the trade off of having two small "main" tests or one that is a bit more complicated I chose the first option. If in the future we will decide to support additional levels/KEMs then making a generic test would be better. With that saying, feel free to change, I am not attached to either choice.